cluster/src/main/scala/io/snappydata/gemxd/SparkSQLExecuteImpl.scala

```diff
@@ -483,7 +483,7 @@ object SnappyContextPerConnection {
     val context = connectionIdMap.get(connectionID)
     if (context != null) context
     else {
-      val snc = SnappyContext.getOrCreate(null: SparkContext).newSession()
+      val snc = SnappyContext(null: SparkContext).newSession()
```

Comment:
I don't understand this reasoning. SparkSession will also be one for every connection. So its behaviour will be exactly the same i.e. one tab per connection unless some separate provision of grouping tabs is made.

> It would mean that for every client connection we will be creating a new tab.

Yes, thats how it should be. Refer to SYS.SESSIONS table in GemXD. Otherwise all queries will get mixed in a single tab making it useless. As I said we can look at clean grouping of tabs later.